docs/end_to_end_tutorials/chatbots/building_a_chatbot.md

```diff
@@ -1,33 +1,44 @@
+In this tutorial, we'll walk you through building a context-augmented chatbot using a [Data Agent](https://gpt-index.readthedocs.io/en/stable/core_modules/agent_modules/agents/root.html). This agent, powered by LLMs, is capable of intelligently executing tasks over your data. The end result is a chatbot agent equipped with a robust set of data interface tools provided by LlamaIndex to answer queries about your data.
 
-**Note**: This is a continuation of some initial work building a query interface over SEC 10-K filings - [check it out here](https://medium.com/@jerryjliu98/how-unstructured-and-llamaindex-can-help-bring-the-power-of-llms-to-your-own-data-3657d063e30d).
+**Note**: TThis tutorial builds upon initial work on creating a query interface over SEC 10-K filings - [check it out here](https://medium.com/@jerryjliu98/how-unstructured-and-llamaindex-can-help-bring-the-power-of-llms-to-your-own-data-3657d063e30d).
 
 ### Context
 
-In this tutorial, we build an "10-K Chatbot" by downloading the raw UBER 10-K HTML filings from Dropbox. The user can choose to ask questions regarding the 10-K filings.
+In this guide, we’ll build a "10-K Chatbot" that uses raw UBER 10-K HTML filings from Dropbox. Users can interact with the chatbot to ask questions related to the 10-K filings.
+
+### Preparation
+
+```python
+import os
+import openai
+
+os.environ["OPENAI_API_KEY"] = "sk-..."
+openai.api_key = os.environ["OPENAI_API_KEY"]
+
+import nest_asyncio
+
+nest_asyncio.apply()
+```
 
 ### Ingest Data
 
-Let's first download the raw 10-k files, from 2019-2022.
+Initially, download the raw 10-K files for the years 2019-2022.
 
 ```python
 # NOTE: the code examples assume you're operating within a Jupyter notebook.
-# download files
+# Create a data directory and download files
 !mkdir data
 !wget "https://www.dropbox.com/s/948jr9cfs7fgj99/UBER.zip?dl=1" -O data/UBER.zip
 !unzip data/UBER.zip -d data
-
 ```
 
-We use the [Unstructured](https://github.com/Unstructured-IO/unstructured) library to parse the HTML files into formatted text.
-We have a direct integration with Unstructured through [LlamaHub](https://llamahub.ai/) - this allows us to convert any text into a Document format that LlamaIndex can ingest.
+To parse the HTML files into formatted text, we use the [Unstructured](https://github.com/Unstructured-IO/unstructured) library. Thanks to [LlamaHub](https://llamahub.ai/), we can directly integrate with Unstructured, allowing conversion of any text into a Document format that LlamaIndex can ingest.
 
 ```python
-
 from llama_index import download_loader, VectorStoreIndex, ServiceContext, StorageContext, load_index_from_storage
```

Comment:
Just a slight preference, but lets import stuff as it's used, rather than in one large block